pkg/credentialprovider/keyring.go

```diff
@@ -121,7 +129,16 @@ func (dk *BasicDockerKeyring) Lookup(image string) (docker.AuthConfiguration, bo
 			continue
 		}
 
-		return dk.creds[k], true
+		dockerAuthKeyMatches = append(dockerAuthKeyMatches, k)
```

Comment:
this changes the behavior to include credentials for a parent path when a more specific set of credentials are available... is that what we want?